fleet/agent_group_service.go

```diff
@@ -80,6 +80,23 @@ func (svc fleetService) ViewAgentGroupByID(ctx context.Context, token string, id
 	return ag, nil
 }
 
+func (svc fleetService) EditAgentGroup(ctx context.Context, token string, group AgentGroup) (AgentGroup, error) {
+	ownerID, err := svc.identify(token)
+	if err != nil {
+		return AgentGroup{}, err
+	}
+
+	if len(group.MatchingAgents) > 0 {
+		return AgentGroup{}, errors.ErrUpdateEntity
+	}
+
+	ag, err := svc.agentGroupRepository.Update(ctx, ownerID, group)
+	if err != nil {
+		return AgentGroup{}, err
+	}
+	return ag, nil
```

Comment:
so updating tags for an AgentGroup will mean any Policies that were using this Agent Group have to be redistributed (and some agents may need to be notified that a policy they were using no longer applies). let's sync up on how to do this